ddtrace/contrib/pytest/plugin.py

```diff
@@ -207,8 +207,8 @@ def pytest_runtest_makereport(item, call):
     else:
         if xfail and not has_skip_keyword:
             # XPass (strict=True) are recorded failed by pytest, longrepr contains reason
-            span.set_tag(test.XFAIL_REASON, result.longrepr)
-            span.set_tag(test.RESULT, test.Status.XPASS.value)
+            span.set_tag(test.XFAIL_REASON, getattr(result, "wasxfail", "XFail"))
+            span.set_tag(test.RESULT, test.Status.XFAIL.value)
```

Comment:
this is a result that failed, so it doesn't make sense to mark it as unexpectedly passing (`XPASS`)